src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala

```diff
@@ -57,9 +81,35 @@ case class MesosCluster(
     // to it's marathon, leading to multiple tasks (from different IT suits) trying to use the same port!
     // First-come-first-served task will bind successfully where the others will fail leading to a lot inconsistency and
     // flakiness in tests.
+
+    val (additionalAttributes: Map[String, Option[String]], faultDomainJson) = if (agentsFaultDomains.nonEmpty && agentsFaultDomains(i).nonEmpty) {
+      val faultDomain = agentsFaultDomains(i).get
+      val faultDomainJson = s"""
+          |{
+          |  "fault_domain":
+          |    {
+          |      "region":
+          |        {
+          |          "name": "${faultDomain.region}"
+          |        },
+          |      "zone":
+          |        {
+          |          "name": "${faultDomain.zone}"
+          |        }
+          |    }
+          |}
+        """.stripMargin
+      Map("fault_domain_region" -> Some(faultDomain.region.value), "fault_domain_zone" -> Some(faultDomain.zone.value)) -> Some(faultDomainJson)
```

Comment:
This is a hard line to read. And `additionalAttributes` are actually `faultDomainAttributes`